src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs

```diff
@@ -76,32 +76,7 @@ internal static DateTime BootTimeToDateTime(TimeSpan timespanAfterBoot)
         }
 
         /// <summary>Gets the system boot time.</summary>
-        private static DateTime BootTime
-        {
-            get
-            {
-                // '/proc/stat -> btime' gets the boot time.
-                // btime is the time of system boot in seconds since the Unix epoch.
-                // It includes suspended time and is updated based on the system time (settimeofday).
-                const string StatFile = Interop.procfs.ProcStatFilePath;
-                string text = File.ReadAllText(StatFile);
-                int btimeLineStart = text.IndexOf("\nbtime ", StringComparison.Ordinal);
-                if (btimeLineStart >= 0)
-                {
-                    int btimeStart = btimeLineStart + "\nbtime ".Length;
-                    int btimeEnd = text.IndexOf('\n', btimeStart);
-                    if (btimeEnd > btimeStart)
-                    {
-                        if (long.TryParse(text.AsSpan(btimeStart, btimeEnd - btimeStart), out long bootTimeSeconds))
-                        {
-                            return DateTime.UnixEpoch + TimeSpan.FromSeconds(bootTimeSeconds);
-                        }
-                    }
-                }
-
-                return DateTime.UtcNow;
-            }
-        }
+        private static DateTime BootTime => DateTime.UnixEpoch + TimeSpan.FromTicks((long)Interop.Sys.GetTimestamp() / 100);
```

Comment:
(Also, #67583 is about StartTime, not BootTime.)

Reply:
The callstack shows that it fails in the BootTime (private method).

Reply:
> The callstack shows that it fails in the BootTime (private method).

Ah, ok, StartTime uses BootTime.

Still, how is the computation you're doing actually computing BootTime?  It seems like it's actually trying to compute the current time, under the assumption that GetTimestamp returns time since the epoch.

Reply:
> I think the name is misleading, but SystemNative_GetTimestamp (CLOCK_MONOTONIC) returns uptime since epoch on linux.

But BootTime is a fixed moment in time, when the system booted.  GetTimestamp returns an ever increasing value each time you call it... it can't be computing a fixed moment in time.  What am I missing?